### PR TITLE
ENG-9789 restructure importer/formatter for isImporterRunEveryWhere

### PR DESCRIPTION
**Pull request by dweiss88**

### CODE REVIEW

**Comment by akhanzode** [comment]
Can you merge from master as you will get conflicts.

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 1]
```diff
@@ -212,7 +221,7 @@ private void log() {
     }
 
     private static Map<URI, KafkaStreamImporterConfig> getConfigsForPartitions(String key, List<HostAndPort> brokerList,
-            final String topic, String groupId, String procedure, int soTimeout, int fetchSize)
+            final String topic, String groupId, String procedure, int soTimeout, int fetchSize, Formatter<String> formatter)
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: How come you use Formatter, Formatter<String>, Formatter<?> shouldn't it be generic in all places?

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 2]
```diff
@@ -137,9 +140,15 @@ public URI getResourceID()
         return m_uri;
     }
 
-    public static Map<URI, ImporterConfig> createConfigEntries(Properties props)
+    @Override
+    public Formatter<String> getFormatter()
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: Formatter<?>?

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 3]
```diff
@@ -137,9 +141,18 @@ public URI getResourceID()
...
+    {
+        return m_formatter;
+    }
+
+    public static Map<URI, ImporterConfig> createConfigEntries(ImportConfiguration config)
     {
-       String brokers = props.getProperty("brokers", "").trim();
+        Properties props = config.getmoduleProperties();
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: Camelcase please `getModuleProperties`

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 4]
```diff
@@ -137,9 +141,18 @@ public URI getResourceID()
...
+        Formatter<String> formatter = (Formatter<String>) config.getFormatter().create(
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: `getFormatterFactory`

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 5]
```diff
@@ -137,9 +141,18 @@ public URI getResourceID()
...
+                config.getFormatName(), config.getformatterProperties());
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: `getFormatterProperties`

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 6]
```diff
@@ -188,7 +201,7 @@ public URI getResourceID()
                 throw new IllegalArgumentException("topic name " + topic + " is illegal, contains a character other than ASCII alphanumerics, '_' and '-'");
             }
             try {
-                configs.putAll(getConfigsForPartitions(key, hapList, topic, groupId, procedure, soTimeout, fetchSize));
+                configs.putAll(getConfigsForPartitions(key, hapList, topic, groupId, procedure, soTimeout, fetchSize, formatter));
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: formatterFactory

File: src/frontend/org/voltdb/importclient/log4j/Log4jSocketImporterConfig.java [hunk 7]
```diff
@@ -77,6 +78,12 @@ public URI getResourceID()
         return m_resourceID;
     }
 
+    @Override
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/log4j/Log4jSocketImporterConfig.java]: this should return the FormatterFactory

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 8]
```diff
@@ -76,6 +79,7 @@ private KafkaStreamImporterConfig(URI uri, List<HostAndPort> brokers, String top
         m_fetchSize = fetchSize;
         m_soTimeout = soTimeout;
         m_procedure = procedure;
+        m_formatter = formatter;
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: formatterFactory

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 9]
```diff
@@ -63,9 +65,10 @@
     private final String m_procedure;
     private final int m_partition;
     private HostAndPort m_partitionLeader;
+    private Formatter<String> m_formatter;
 
     private KafkaStreamImporterConfig(URI uri, List<HostAndPort> brokers, String topic, int partition, HostAndPort partitionLeader,
-            String groupId, int fetchSize, int soTimeout, String procedure)
+            String groupId, int fetchSize, int soTimeout, String procedure, Formatter<String> formatter)
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: formatterFactory

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java (hunk 2 above)
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: getFormatterFactory

**Pull request closed without merging**